### ISSUE TITLE
"Domain" required for authentication on certain devices

### ISSUE THREAD

**Issue by sunnykoh** [zerotypic/wasg-register#34]
Using the script on Linux but when using the userid/password on Chromebook, new requirement of "domain" is needed as an additional field. New method at https://go.gov.sg/wsgx  generates all 3 fields required similar to script for linux under non-sim card request.

**Comment by zerotypic** [comment]
Hmm, does the site you linked above basically allow you to complete the registration just using the website? In which case, `wasg-register` would not be needed anymore, right? That would be a good move by IMDA if so, glad to see them provide a way to register for people using other kinds of devices.

**Comment by sunnykoh** [comment]
I fully agree, but maybe an update to the readme or help text would be helpful. Surprisingly, most people will google this place first instead of the IMDA site. (Search terms like Linux and Wireless@SG.) I have Been using your Python script for many years.

**Comment by zerotypic** [comment]
Sure, I just need to confirm that the new website is good enough to supersede this script. Not able to test this out right now. I guess it worked for you? Do you foresee  any situation where the website wouldn't suffice and the script would still be needed?

**Comment by sunnykoh** [comment]
I have yet to test the codes as I have not been to a Wireless@SGX location. The issue is that IDA removed this feature for many years (From their website) and your script was the go around.  I hope the IMDA site will stay up because unless things change for Chromebooks which MOE is pushing can be provisioned with Wireless@SGX thru other methods. I suspect the it will stay up but went policies change, it might disappear again.

**Comment by harishpillay** [comment]
This is an interesting development. I will try to get to a WSG location with my laptop (usually a Library) this weekend to check if my current settings are working.

**Comment by zerotypic** [comment]
Ok thanks! Let me know what you find out!

**Comment by JohnathanLoh** [comment]
Also encountered the issue on a ChromeOS device.

Managed to find a quick roundabout solution in this url on IMDA's website: https://eservice.imda.gov.sg/wirelessSGx/

I did not check for the others, but since the singtel domain is "singtel-wsg.singtel.com", likely the other domains will be something similar/fixed.

**Comment by sunnykoh** [comment]
I got rinoa.m1net.com.sg for M1 using the form during my run.

**Comment by harishpillay** [comment]
![Image](https://github.com/user-attachments/assets/0b62c515-21b8-452e-9d88-14746f740600)

I finally got to a Wireless@SGx location - just inside Suntec City, next to the busstop. 

Went to the https://eservice.imda.gov.sg/wirelessSGx/, picked the provider etc and got the credentials. 

See the screenshot. 

All's well now. 

I did a screen recording of what I did and will post it in my blog (harishpillay.com) later along with a link to this issue.

I'm running Fedora 41 btw.

**Comment by sunnykoh** [comment]
So far we have

SingTel : singtel-wsg.singtel.com
M1 : rinoa.m1net.com.sg
SIMBA / TPG : tpgmobile.sg

I think we are missing StarHub domain

**Comment by harishpillay** [comment]
Starhub: uniaaa2.wifi.starhub.net.sg is the domain.

**Comment by harishpillay** [comment]
you can just go to https://eservice.imda.gov.sg/wirelessSGx/, register again and pick the telcos.

**Comment by harishpillay** [comment]
https://harishpillay.com/2025/01/20/wirelesssgx-for-linux-2025-update/ posted.

**Comment by zerotypic** [comment]
Ok thanks for checking it out @harishpillay! I'll update the README to reflect this change, and point people towards your blog post.
